tests/src/test/scala/org/http4s/parser/AuthorizationHeaderSpec.scala

```diff
@@ -15,12 +15,19 @@ class AuthorizationHeaderSpec extends Http4sSpec {
       hparse(h.value) must be_\/-(h)
     }
 
-    "Reject an ivalid Oauth2 header" in {
+    "Reject an invalid Oauth2 header" in {
       val invalidTokens = Seq("f!@", "=abc", "abc d")
       forall(invalidTokens) { token =>
         val h = Authorization(OAuth2BearerToken(token))
         hparse(h.value) must be_-\/
       }
     }
+
+    "Parse a GenericCredential header" in {
+      val scheme = "token"
+      val token = "adsfafsf2332fasdad322332"
+      val h = Authorization(GenericCredentials(scheme.ci, token))
```

Comment:
What happens if you create a `KeyValueCredentials` and try to parse it back?  I bet you get a `GenericCredentials`.  The parser can tell from the scheme which concrete type to parse.  Now there are two types without a specific scheme.

I think maybe we keep your `GenericCredentials` and bid farewell to the map-based generic credentials.  What do you think?